pkg/progress/progress.go

```diff
@@ -83,18 +82,28 @@ func (m *Manager) UpdateProgressRemaining(progress string, remaining float64) {
 		if p.total < remaining {
 			p.total = remaining
 		}
-		if p.lastTimeRemaining < remaining {
-			p.lastTimeRemaining = remaining
+
+		// We use window size / update interval to get the history length.
+		historyLen := int(speedStatisticalWindow / updateInterval)
+		if len(p.history) > historyLen {
+			p.history = p.history[1:]
```

Comment:
Would this cause a GC problem? It seems once meet the historyLen, this operator will be done as frequently as the `UpdateProgress` called?  How about use a slice with constant length to implement the FIFOQueue?  For example, we assume the `historyLength` keep 5, and there is a slice::
1,2,3,4,5 (first=0,last:4)
when 6 comes, the slice changed to
6,2,3,4,5 (first=1,last=0)
and 7 comes: 
6,7,3,4,5 (first=2,last=1)